test/refunder-test.ts

```diff
@@ -0,0 +1,295 @@
+
+			const resAfter = await refunder.refundables(randomAddress, randomFuncIdAsBytes); 
+			expect(resAfter).to.be.eq(true);
+		});
+
+		it("Owner should be able to edit refundable", async function() {
+		
+			const randomAddress = greeter.address;
+			const randomFuncId = ethers.utils.id('setGreeting(string)');
+			const randomFuncIdAsBytes = ethers.utils.arrayify(randomFuncId.substr(0, 10))
+
+			// address, bytes4
+			const resBefore = await refunder.refundables(randomAddress, randomFuncIdAsBytes); 
+			expect(resBefore).to.be.eq(false);
+
+			let res = await refunder.whitelistRefundable(randomAddress, randomFuncIdAsBytes, true);
+			await res.wait();
+
+			const resAfter = await refunder.refundables(randomAddress, randomFuncIdAsBytes); 
+			expect(resAfter).to.be.eq(true);
+
+			res = await refunder.whitelistRefundable(randomAddress, randomFuncIdAsBytes, false);
+			await res.wait();
+
+			const resAfterEdit = await refunder.refundables(randomAddress, randomFuncIdAsBytes); 
+			expect(resAfterEdit).to.be.eq(false);
+		});
+	
+		it("Not Owner should be able to check for refundable", async function() {
+			const randomAddress = greeter.address;
+			const randomFuncId = ethers.utils.id('setGreeting(string)');
+			const randomFuncIdAsBytes = ethers.utils.arrayify(randomFuncId.substr(0, 10))
+
+			// address, bytes4
+			const res = await refunder.connect(addr1).refundables(randomAddress, randomFuncIdAsBytes); 
+			
+			expect(res).to.be.eq(false);
+		});
+
+		it("Not Owner should NOT be able to add, edit refundable", async function() {
+		
+			const randomAddress = greeter.address;
+			const randomFuncId = ethers.utils.id('setGreeting(string)');
+			const randomFuncIdAsBytes = ethers.utils.arrayify(randomFuncId.substr(0, 10))
+
+			await expect(refunder.connect(addr1).whitelistRefundable(randomAddress, randomFuncIdAsBytes, true)).to.be.revertedWith(NOT_AN_OWNER);
+		});
+	});
+
+	describe('Relay and refund', () => {
+
+		it('Not owner should successfully call refunder(whitelisted)', async () => {
+
+			const greeterAddress = greeter.address;
+			const funcId = ethers.utils.id('setGreeting(string)');
+			const funcIdAsBytes = ethers.utils.arrayify(funcId.substr(0, 10));
+
+			let res = await owner.sendTransaction({
+				value: ethToWei("1"),
+				to: refunder.address
+			});
+
+			let txReceipt = await res.wait();
+			
+			res = await refunder.whitelistRefundable(greeterAddress, funcIdAsBytes, true);
+			await res.wait();
+
+			res = await refunder.setMaxGasPrice('150000000000'); // 150 gwei
+			await res.wait();
+
+			const balanceBefore = await ethers.provider.getBalance(addr1.address);
+			
+			const text = 'Hello, Tester!';
+			const hexString = ethers.utils.formatBytes32String(text);
+			const args = ethers.utils.arrayify(hexString);
+
+			res = await refunder.connect(addr1).relayAndRefund(greeterAddress, funcIdAsBytes, args);
+			txReceipt = await res.wait();
+
+			const balanceAfter = await ethers.provider.getBalance(addr1.address);
```

Comment:
We should assert balances after the call & assert the effect of the function -> call the greeter and check that the call was forwarded